File: Web/src/player/states/RunningState.ts
import { Ticker } from "pixi.js";
import IPlayerState from "./IPlayerState.ts";
import IPlayerStateContext from "./IPlayerStateContext.ts";

class RunningState implements IPlayerState {
  private _wantToJump = false;

  constructor(
    private readonly _player: IPlayerStateContext,
    private readonly _getJumpingState: () => IPlayerState,
  ) {}

  enter() {
    this._wantToJump = false;
    this._player.onGround = true;
    this._player.setCurrentAnimation(this._player.animations.run);
  }

  handleJump() {
    this._wantToJump = true;
  }

  update(_time: Ticker): IPlayerState {
    if (this._wantToJump) return this._getJumpingState();
    return this;
  }
}

export default RunningState;

File: CSharp/Starfall/Players/States/RunningState.cs
using Microsoft.Xna.Framework;
using Starfall.Assets;
using System;

namespace Starfall.Players.States
{
    public class RunningState : IPlayerState
    {
        private bool _wantToJump = false;
        private readonly Rectangle _collisionThreashold = new Rectangle(45, 38, 75, 80);
        private readonly Player _player;
        private readonly StatesManager _statesManager;

        public RunningState(
            Player player,
            StatesManager statesManager)
        {
            _player = player;
            _statesManager = statesManager;
        }

        public void Enter()
        {
            _wantToJump = false;
            _player.OnGround = true;
            _player.AnimationsManager.PlayAnimation(AssetsLoader.AnimationsNames.PlayerRun.ToString());
            _player.DrawingInfos.HitBoxTolerance = _collisionThreashold;
            _player.BodyGlowOffset = new Vector2(
                13 + -_player.CurrentAnimation.CurrentFrameWidth / 2, 
                -_player.CurrentAnimation.CurrentFrameHeight / 2);
        }

        public void HandleJump()
            => _wantToJump = true;

        public IPlayerState Update(TimeSpan elapsed)
        {
            if (_wantToJump)
            {
                return _statesManager.JumpingState;
            }
            
            return this;
        }

    }
}
